lib/Reader/readerir.cpp

```diff
@@ -6543,6 +6543,40 @@ IRNode *GenIR::loadAndBox(CORINFO_RESOLVED_TOKEN *ResolvedToken, IRNode *Addr,
   return box(ResolvedToken, Value);
 }
 
+IRNode *GenIR::makeRefAny(CORINFO_RESOLVED_TOKEN *ResolvedToken,
+                          IRNode *Object) {
+
+  // Create a new temporary of the right type.
+  CORINFO_CLASS_HANDLE RefAnyHandle = getBuiltinClass(CLASSID_TYPED_BYREF);
+  Type *RefAnyTy = getType(CORINFO_TYPE_VALUECLASS, RefAnyHandle);
+  assert(RefAnyTy->isStructTy() && "expect struct type for refany");
```

Comment:
This assert looks redundant both because getType always returns StructTypes for CORINFO_TYPE_VALUECLASS and because you are casting on the next line.